From 9fa0a814931a1c25bc50b3d67b6fb7c5d641c85f Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <jwilsdon@joyent.com>
Date: Wed, 13 Mar 2019 18:22:40 -0700
Subject: [PATCH] TRITON-1306 Setup when specifying a hostname was
 inadvertently broken Reviewed by: Orlando Vazquez <orlando@joyent.com>
 Approved by: Orlando Vazquez <orlando@joyent.com>

---
 lib/models/server.js | 8 ++++----
 package.json         | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/models/server.js b/lib/models/server.js
index a0c00a0..71ae8aa 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -1147,8 +1147,8 @@ ModelServer.prototype.setup = function (params, callback) {
             wfParams.disk_layout = params.disk_layout;
         }
 
-        async.waterfall([
-            function (cb) {
+        vasync.pipeline({funcs: [
+            function _optionallySetHostname(_, cb) {
                 if (params.hasOwnProperty('hostname') && params.hostname) {
                     ModelServer.upsert(self.uuid, {hostname: params.hostname}, {
                         etagRetries: 0
@@ -1156,7 +1156,7 @@ ModelServer.prototype.setup = function (params, callback) {
                 } else {
                     cb();
                 }
-            }, function (cb) {
+            }, function _instantiateSetupWorkflow(_, cb) {
                 self.log.info('Instantiating server-setup workflow');
                 ModelServer.getWorkflow().getClient().createJob(
                     'server-setup',
@@ -1173,7 +1173,7 @@ ModelServer.prototype.setup = function (params, callback) {
                         return;
                     });
             }
-        ], function (err) {
+        ]}, function (err) {
             callback(err, job_uuid);
         });
     });
diff --git a/package.json b/package.json
index 8249d14..c574db9 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.21.3",
+  "version": "1.21.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

